api/conftest.py

```diff
@@ -167,15 +174,28 @@ def django_db_setup(request: pytest.FixtureRequest) -> None:
         # Django's test database clone indices start at 1,
         # Pytest's worker indices are 0-based
         test_db_suffix = str(int(xdist_worker_id_suffix) + 1)
+
+        from django.conf import settings
+
+        for db_settings in settings.DATABASES.values():
+            test_db_name = (
+                f"{TEST_DATABASE_PREFIX}{db_settings['NAME']}_{test_db_suffix}"
+            )
+            db_settings["NAME"] = test_db_name
+
     else:
         # Tests are run on main node, which assumes -n0
-        return request.getfixturevalue("django_db_setup")  # type: ignore[no-any-return] # pragma: no cover
+        request.getfixturevalue("django_db_setup")  # pragma: no cover
 
-    from django.conf import settings
+    if request.config.option.ci or request.config.getoption("reuse_db"):
+        from django.db import connections
 
-    for db_settings in settings.DATABASES.values():
-        test_db_name = f"{TEST_DATABASE_PREFIX}{db_settings['NAME']}_{test_db_suffix}"
-        db_settings["NAME"] = test_db_name
+        # Ensure psycopg type handlers are registered.
+        # This is necessary for tests that involve `HStoreField`.
+        for connection in connections.all():
+            if connection.vendor == "postgresql":
+                with django_db_blocker.unblock():
+                    register_type_handlers(connection)
```

Comment:
Pretty yucky! Do we have something we can reference here as to why this is necessary?